src/allmydata/test/web/test_introducer.py

```diff
@@ -222,16 +233,12 @@ def test_json(self):
             0,
         )
 
-        resource = IntroducerRoot(introducer_node)
-        response = json.loads(
-            self.successResultOf(
+        resource = IntroducerRoot(introducer_node) 
+        response = json.loads(succeeded(
                 render(resource, {b"t": [b"json"]}),
-            ),
-        )
-        self.assertEqual(
-            response,
-            {
+            )._matcher.result)
+        deferred = defer.succeed({
                 u"subscription_summary": {"arbitrary": 2},
                 u"announcement_summary": {"arbitrary": 1},
-            },
-        )
+            })
+        self.assertThat(deferred.result, Equals(response))
```

Comment:
The `result` attribute of `Deferred` is _not_ marked as private by the `_` prefix convention ... but it is not good practice to use it, either.  Fortunately I think correct use of `succeeded` will remove the need for this attribute.